packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1319,6 +1340,8 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     return _layer;
   }
 
+  /// Whether this render object requires an update to its compositing state.
+  bool get needsCompositingBitsUpdate => _needsCompositingBitsUpdate;
```

Comment:
needsPaint looks easy to remove.  Folks (including me) are already using needsLayout for some dubious things...